trainlib/viewer.py

```diff
@@ -198,23 +214,32 @@ def _create_image_box(self, figsize):
 
         return image_box
 
-    def _generate_views(self):
-        image_box = self._create_image_box(self.figsize)
+    def _ensure_correct_dims(self, array: np.ndarray) -> np.ndarray:
+        array = np.squeeze(array)
+        if array.ndim == 2:
+            array = np.expand_dims(array, 0)  # add channel
+        if array.ndim != 3:
+            raise ValueError(f"Input array needs to be 2d or 3d, but is {array.ndim}d")
+        return array
```

Comment:
Everything subclasses or uses BasicViewer. Making it 2D/3D also creates 2D/3D classes for DICOM explorer and for ListViewer and for show_batch. 

I do not think this would be worth it or make the cod more readable.